dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnableTest.kt

```diff
@@ -207,6 +207,101 @@ internal class DataUploadRunnableTest {
         )
     }
 
+    @ParameterizedTest
+    @EnumSource(
+        SystemInfo.BatteryStatus::class,
+        mode = EnumSource.Mode.INCLUDE
+    )
+    fun `M send batch W device has external power source`(
+        batteryStatus: SystemInfo.BatteryStatus,
+        @Forgery batch: Batch,
+        forge: Forge
+    ) {
+        val systemInfo = SystemInfo(
+            batteryStatus = batteryStatus,
+            batteryLevel = forge.anInt(0, 100),
```

Comment:
Here we don't test that a battery level of 0 actually works (I mean we do once every 100 test). Maybe we should ensure that even with a very low battery, being plugged in always works. 
```suggestion
            batteryLevel = forge.anInt(0, 5),
```

Reply:
Yeah I am still correcting those tests that's why I marked it as a draft ;)